app/scripts/lib/rpc-method-middleware/handlers/wallet-createSession/handler.js

```diff
@@ -0,0 +1,218 @@
+
+    const validScopedProperties = processScopedProperties(
+      flattenedRequiredScopes,
+      flattenedOptionalScopes,
+      scopedProperties,
+    );
+
+    const existsNetworkClientForChainId = (chainId) => {
+      try {
+        hooks.findNetworkClientIdByChainId(chainId);
+        return true;
+      } catch (err) {
+        return false;
+      }
+    };
+
+    const existsEip3085ForChainId = (chainId) => {
+      const scopeString = `eip155:${parseInt(chainId, 16)}`;
+      return validScopedProperties?.[scopeString]?.eip3085;
+    };
+
+    const {
+      supportedScopes: supportedRequiredScopes,
+      supportableScopes: supportableRequiredScopes,
+      unsupportableScopes: unsupportableRequiredScopes,
+    } = bucketScopes(flattenedRequiredScopes, {
+      isChainIdSupported: existsNetworkClientForChainId,
+      isChainIdSupportable: existsEip3085ForChainId,
+    });
+
+    const {
+      supportedScopes: supportedOptionalScopes,
+      supportableScopes: supportableOptionalScopes,
+      unsupportableScopes: unsupportableOptionalScopes,
+    } = bucketScopes(flattenedOptionalScopes, {
+      isChainIdSupported: existsNetworkClientForChainId,
+      isChainIdSupportable: existsEip3085ForChainId,
+    });
+
+    // TODO: placeholder for future CAIP-25 permission confirmation call
+    JSON.stringify({
+      supportedRequiredScopes,
+      supportableRequiredScopes,
+      unsupportableRequiredScopes,
+      supportedOptionalScopes,
+      supportableOptionalScopes,
+      unsupportableOptionalScopes,
+    });
+
+    // These should be EVM accounts already although the name does not necessary imply that
+    // These addresses are lowercased already
+    const existingEvmAddresses = hooks
+      .listAccounts()
+      .map((account) => account.address);
+    const supportedEthAccounts = getEthAccounts({
+      requiredScopes: supportedRequiredScopes,
+      optionalScopes: supportedOptionalScopes,
+    })
+      .map((address) => address.toLowerCase())
+      .filter((address) => existingEvmAddresses.includes(address));
+    const supportedEthChainIds = getPermittedEthChainIds({
+      requiredScopes: supportedRequiredScopes,
+      optionalScopes: supportedOptionalScopes,
+    });
+
+    const legacyApproval = await hooks.requestPermissionApprovalForOrigin({
+      [PermissionNames.eth_accounts]: {
+        caveats: [
+          {
+            type: CaveatTypes.restrictReturnedAccounts,
+            value: supportedEthAccounts,
+          },
+        ],
+      },
+      [PermissionNames.permittedChains]: {
+        caveats: [
+          {
+            type: CaveatTypes.restrictNetworkSwitching,
+            value: supportedEthChainIds,
+          },
+        ],
+      },
+    });
+
+    let caip25CaveatValue = {
+      requiredScopes: supportedRequiredScopes,
+      optionalScopes: supportedOptionalScopes,
+      isMultichainOrigin: true,
+      // TODO: preserve sessionProperties?
```

Comment:
thoughts? we aren't actually using them currently after the session is actually created